public/javascripts/songForm.js

```diff
@@ -0,0 +1,264 @@
+
+    $(window).load(function() {
+        if (song.title === '') {
+            $('#copypaste').modal('show');
+        }
+    });
+
+    function addSong() {
+        m.request({
+                method: 'post',
+                url: url,
+                data: {
+                    title: title(),
+                    author: author(),
+                    year: year(),
+                    lang: lang(),
+                    copyright: copyright(),
+                    lyric: lyric
+                }
+            })
+            .then(function(res) {
+                if (res.errorMessages) {
+                    errors = res.errorMessages;
+                    $('body').scrollTop(0);
+                } else {
+                    window.location.href = res.url
+                }
+            })
+    }
+    var addStanza = {
+        view: function(ctrl, args) {
+            return m('.form-group', [
+                m('label', 'Stanza ' + (args.index + 1) + ':'),
+                m('textarea.form-control', {
+                    value: function() {
+                        return args.stanza.reduce((prev, curr) => {
+                            return prev + '\n' + curr
+                        })
+                    }(),
+                    rows: '5',
+                    config: function(elem, isInit) {
+                        if (!isInit) {
+                            if (readonly) {
+                                $(elem).prop('readonly', true)
+                            }
+                            $(elem).bind('input propertychange', function() {
+                                var newStanza = $(elem).val() ? $(elem).val().split(/\r?\n|\//) : ''
+                                lyric.splice(args.index, 1, newStanza)
+                            });
+                        }
+                    }
+                }),
+                m('button.btn.btn-default', {
+                    disabled: args.length === 1 ? true : false,
+                    onclick: function() {
+                        lyric.splice(args.index, 1)
+                    }
+                }, [
+                    m('span.glyphicon.glyphicon-minus')
+                ]),
+                m('button.btn.btn-default', {
+                    onclick: function() {
+                        lyric.splice(args.index + 1, 0, [''])
+                    }
+                }, [
+                    m('span.glyphicon.glyphicon-plus')
+                ])
+            ])
+        }
+    }
+
+    function copypasteModal() {
+        return m('#copypaste.modal.fade[role=dialog]', [
+            m('.modal-dialog', [
+                m('.modal-content', [
+                    m('.modal-body', [
+                        m('label', 'Lyric:'),
+                        m('textarea.form-control', {
+                            rows: '25',
+                            placeholder: 'Write the lyric with new line as a separator between stanzas',
```

Comment:
See my earlier comment. If this is exactly the same as above, turn this into a reusable component instead of duplicating code.